core/controllers/admin_test.py

```diff
@@ -85,6 +142,430 @@ def test_change_configuration_property(self):
+        response = self.get_html_response('/admin')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.assertRaisesRegexp(
+            Exception, 'invalid_type is not a number'):
+            self.post_json(
+                '/adminhandler', {
+                    'action': 'generate_dummy_explorations',
+                    'num_dummy_exps_to_publish': 1,
+                    'num_dummy_exps_to_generate': 'invalid_type'
+                }, csrf_token=csrf_token)
+
+        self.logout()
+
+    def test_handler_raises_error_with_non_int_num_dummy_exps_to_publish(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        response = self.get_html_response('/admin')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.assertRaisesRegexp(
+            Exception, 'invalid_type is not a number'):
+            self.post_json(
+                '/adminhandler', {
+                    'action': 'generate_dummy_explorations',
+                    'num_dummy_exps_to_publish': 'invalid_type',
+                    'num_dummy_exps_to_generate': 1
+                }, csrf_token=csrf_token)
+
+        self.logout()
+
+    def test_cannot_generate_dummy_explorations_in_prod_mode(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        response = self.get_html_response('/admin')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        prod_mode_swap = self.swap(constants, 'DEV_MODE', False)
+        assert_raises_regexp_context_manager = self.assertRaisesRegexp(
+            Exception, 'Cannot generate dummy explorations in production.')
+
+        with assert_raises_regexp_context_manager, prod_mode_swap:
+            self.post_json(
+                '/adminhandler', {
+                    'action': 'generate_dummy_explorations',
+                    'num_dummy_exps_to_generate': 10,
+                    'num_dummy_exps_to_publish': 3
+                }, csrf_token=csrf_token)
+
+        self.logout()
+
+    def test_admin_topics_csv_download_handler(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        response = self.get_custom_response(
+            '/admintopicscsvdownloadhandler', expected_content_type='text/csv')
+
+        self.assertEqual(
+            response.headers['Content-Disposition'],
+            'attachment; filename=topic_similarities.csv')
+
+        self.logout()
+
+    def test_admin_job_output_handler(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+
+        self.save_new_valid_exploration('exp_id', self.admin_id)
+
+        job_id = SampleMapReduceJobManager.create_new()
+        SampleMapReduceJobManager.enqueue(job_id)
+
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+
+        response = self.get_json('/adminjoboutput', params={'job_id': job_id})
+
```

Comment:
I suggest dropping this newline; also drop the one on line 296. Helps connect the response to the check.

(If you have too many one-liner blocks, it becomes similarly hard to read as if you have no blocks at all.)

Reply:
done